include/sys/fs/zfs.h

```diff
@@ -543,6 +544,12 @@ typedef enum zfs_key_location {
 	ZFS_KEYLOCATION_LOCATIONS
 } zfs_keylocation_t;
 
+typedef enum {
+	ZFS_PREFETCH_ALL = 0,
+	ZFS_PREFETCH_METADATA = 1,
+	ZFS_PREFETCH_NONE = 2
+} zfs_prefetch_type_t;
```

Comment:
It is similar to zfs_cache_type_t, but for some reason in different order.  I think if we could reuse that one or prefer duplication, but matching the same order at least would be good.

Reply:
Yeah, I would avoid it. However, during pool import `os_prefetch` defaults to 0. If we leave `ZFS_PREFETCH_NONE` as the first entry, it will be enumerated as 0 and prefetch would be disabled for pool import (causing slower import).

Reply:
Looking on os_primary_cache as example, I see dmu_objset_open_impl() initalizing it immediately either from property or from a constant.  What scenario of slower import are you talking about, where os_prefetch is needed before that?

Reply:
You are right, I can set a default in `dmu_objset_open_impl()`. I completely missed it, thanks. Now the enum is in the right order: NONE, METADATA, ALL.